### PR TITLE
[MIG] Port base_user_role to 10.0

### PR DESCRIPTION
**Pull request by adrienpeiffer**

### CODE REVIEW

File: base_user_role/tests/test_user_role.py [hunk 1]
```diff
@@ -4,104 +4,92 @@
...
-            self.cr, self.uid,
             {'name': u"USER TEST (ROLES)", 'login': 'user_test_roles'})
 
         # ROLE_1
-        self.group_user_id = self.imd_model.get_object_reference(
-            self.cr, self.uid, 'base', 'group_user')[1]
-        self.group_no_one_id = self.imd_model.get_object_reference(
-            self.cr, self.uid, 'base', 'group_no_one')[1]
+        self.group_user_id = self.env.ref('base.group_user')
+        self.group_no_one_id = self.env.ref('base.group_no_one')
```
Comment (sebalix) [inline base_user_role/tests/test_user_role.py]: s/group_user_id/group_user/
s/group_no_one_id/group_no_one/

File: base_user_role/tests/test_user_role.py [hunk 2]
```diff
@@ -4,104 +4,92 @@
...
+        self.role1_id = self.role_model.create(vals)
 
         # ROLE_2
-        self.group_multi_currency_id = self.imd_model.get_object_reference(
-            self.cr, self.uid, 'base', 'group_multi_currency')[1]
-        self.group_sale_manager_id = self.imd_model.get_object_reference(
-            self.cr, self.uid, 'base', 'group_sale_manager')[1]
+        self.group_multi_currency_id = self.env.ref(
+            'base.group_multi_currency')
+        self.group_settings_id = self.env.ref('base.group_system')
```
Comment (sebalix) [inline base_user_role/tests/test_user_role.py]: s/group_settings_id/group_settings/

File: base_user_role/tests/test_user_role.py [hunk 3]
```diff
@@ -4,104 +4,92 @@
...
 
     def setUp(self):
         super(TestUserRole, self).setUp()
-        self.imd_model = self.registry('ir.model.data')
-        self.user_model = self.registry('res.users')
-        self.role_model = self.registry('res.users.role')
+        self.user_model = self.env['res.users']
+        self.role_model = self.env['res.users.role']
 
         self.user_id = self.user_model.create(
```
Comment (sebalix) [inline base_user_role/tests/test_user_role.py]: s/user_id/user/

File: base_user_role/tests/test_user_role.py [hunk 4]
```diff
@@ -4,104 +4,92 @@
...
-                (0, 0, {'role_id': self.role2_id, 'date_to': yesterday_str}),
+                (0, 0,
+                 {'role_id': self.role2_id.id, 'date_to': yesterday_str}),
             ]})
-        user = self.user_model.browse(self.cr, self.uid, self.user_id)
-        user_group_ids = sorted(set([group.id for group in user.groups_id]))
-        role1_group_ids = role1.implied_ids.ids
-        role1_group_ids.append(role1.group_id.id)
+        user_group_ids = sorted(set(
+            [group.id for group in self.user_id.groups_id]))
```
Comment (sebalix) [inline base_user_role/tests/test_user_role.py]:
```python
user_group_ids = sorted(set(self.user.groups_id.ids))
```

File: base_user_role/tests/test_user_role.py [hunk 5]
```diff
@@ -4,104 +4,92 @@
...
-            self.cr, self.uid, 'base', 'group_user')[1]
-        self.group_no_one_id = self.imd_model.get_object_reference(
-            self.cr, self.uid, 'base', 'group_no_one')[1]
+        self.group_user_id = self.env.ref('base.group_user')
+        self.group_no_one_id = self.env.ref('base.group_no_one')
         vals = {
             'name': u"ROLE_1",
-            'implied_ids': [6, 0, [self.group_user_id, self.group_no_one_id]],
+            'implied_ids': [6, 0, [self.group_user_id.id,
+                                   self.group_no_one_id.id]],
```
Comment (sebalix) [inline base_user_role/tests/test_user_role.py]:
```python
[self.group_user.id, self.group_no_one.id]
```

File: base_user_role/tests/test_user_role.py [hunk 6]
```diff
@@ -4,104 +4,92 @@
...
         }
-        self.role1_id = self.role_model.create(self.cr, self.uid, vals)
+        self.role1_id = self.role_model.create(vals)
```
Comment (sebalix) [inline base_user_role/tests/test_user_role.py]: s/role1_id/role1/

File: base_user_role/tests/test_user_role.py [hunk 7]
```diff
@@ -4,104 +4,92 @@
...
+        self.group_settings_id = self.env.ref('base.group_system')
         vals = {
             'name': u"ROLE_2",
-            'implied_ids': [6, 0, [self.group_multi_currency_id,
-                                   self.group_sale_manager_id]],
+            'implied_ids': [6, 0, [self.group_multi_currency_id.id,
+                                   self.group_settings_id.id]],
         }
-        self.role2_id = self.role_model.create(self.cr, self.uid, vals)
+        self.role2_id = self.role_model.create(vals)
```
Comment (sebalix) [inline base_user_role/tests/test_user_role.py]: s/role2_id/role2/

File: base_user_role/tests/test_user_role.py [hunk 8]
```diff
@@ -4,104 +4,92 @@
...
-            self.cr, self.uid, [self.user_id],
-            {'role_line_ids': [(0, 0, {'role_id': self.role2_id})]})
-        user = self.user_model.browse(self.cr, self.uid, self.user_id)
-        user_group_ids = sorted(set([group.id for group in user.groups_id]))
-        role_group_ids = role2.implied_ids.ids
-        role_group_ids.append(role2.group_id.id)
+        self.user_id.write(
+            {'role_line_ids': [(0, 0, {'role_id': self.role2_id.id})]})
+        user_group_ids = sorted(set(
+            [group.id for group in self.user_id.groups_id]))
```
Comment (sebalix) [inline base_user_role/tests/test_user_role.py]:
```python
user_group_ids = sorted(set(self.user_id.groups_id.ids))

File: base_user_role/tests/test_user_role.py [hunk 9]
```diff
@@ -4,104 +4,92 @@
...
-            self.cr, self.uid, [self.user_id],
-            {'role_line_ids': [(0, 0, {'role_id': self.role1_id})]})
-        user = self.user_model.browse(self.cr, self.uid, self.user_id)
-        user_group_ids = sorted(set([group.id for group in user.groups_id]))
-        role_group_ids = role1.implied_ids.ids
-        role_group_ids.append(role1.group_id.id)
+        self.user_id.write(
+            {'role_line_ids': [(0, 0, {'role_id': self.role1_id.id})]})
+        user_group_ids = sorted(set(
+            [group.id for group in self.user_id.groups_id]))
```
Comment (sebalix) [inline base_user_role/tests/test_user_role.py]:
```python
user_group_ids = sorted(set(self.user.groups_id.ids))
```

**Review by sebalix — APPROVED** [review]
Thank you for the migration! Minor changes to the unit tests about variable names and list comprehensions to remove, but non-blocking. :+1:

File: base_user_role/i18n/am.po [hunk 10]
```diff
@@ -0,0 +1,207 @@
+# Translation of Odoo Server.
+# This file contains the translation of the following modules:
+# * base_user_role
+# 
+# Translators:
+# OCA Transbot <transbot@odoo-community.org>, 2016
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 9.0c\n"
```
Comment (jcdrubay) [inline base_user_role/i18n/am.po]: Seems like it is a wrong version here.

File: base_user_role/views/role.xml [hunk 11]
```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2014 ABF OSIELL <http://osiell.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+<record id="view_res_users_role_form" model="ir.ui.view">
+    <field name="name">res.users.role.form</field>
+    <field name="model">res.users.role</field>
+    <field name="arch" type="xml">
+        <form string="Role" version="7.0">
```
Comment (jcdrubay) [inline base_user_role/views/role.xml]: It should be time to remove the version="7.0"

**Review by jcdrubay — APPROVED** [review]
Only 2 small comments. Thanks for the contribution. I hope it can be merged soon. :)

File: base_user_role/i18n/am.po (hunk 10 above)
Comment (lasley) [reply base_user_role/i18n/am.po]: I'm fairly certain this won't be an issue. Something something, automation is amazing

**Review by lasley — APPROVED** [review]
Thanks @adrienpeiffer 

Can you please remove the version as noted by @jcdrubay & squash the commits? Good for merge after those tiny details 👍

**Comment by adrienpeiffer** [comment]
@lasley @jcdrubay Thanks for the review !

**Comment by lasley** [comment]
@adrienpeiffer - Dang, looks like the last round of edits made CI 🔴  - https://travis-ci.org/OCA/server-tools/jobs/200071164#L720

**Comment by adrienpeiffer** [comment]
It should be better now :green_heart:

**Comment by rousseldenis** [comment]
Ready to be merged ?

**Comment by adrienpeiffer** [comment]
GO :smiley:

**Pull request merged**